apiserver/src/main/java/org/dependencytrack/tasks/maintenance/WorkflowMaintenanceTask.java

```diff
@@ -107,6 +121,25 @@ private Statistics informLocked(final Handle jdbiHandle) {
                 return;
             }
 
+            // Send BOM_PROCESSING_FAILED notifications for timed out failures.
+            final var notificationSubjectDao = jdbiHandle.attach(NotificationSubjectDao.class);
+            final List<BomProcessingFailedSubject> notificationSubjectsForFailure =
+                    notificationSubjectDao.getForBomProcessingTimedOut(failedStepIds);
+            final var notifications = new ArrayList<KafkaEvent<?, ?>>(notificationSubjectsForFailure.size());
+            notificationSubjectsForFailure.stream()
+                    .map(subject -> Notification.newBuilder()
+                            .setScope(SCOPE_PORTFOLIO)
+                            .setGroup(GROUP_BOM_PROCESSING_FAILED)
+                            .setLevel(LEVEL_INFORMATIONAL)
+                            .setTimestamp(Timestamps.now())
+                            .setTitle(NotificationConstants.Title.BOM_PROCESSING_FAILED)
+                            .setContent("A %s BOM processing failed".formatted(subject.getBom().getFormat()))
+                            .setSubject(Any.pack(subject))
+                            .build())
+                    .map(KafkaEventConverter::convert)
+                    .forEach(notifications::add);
+            eventDispatcher.dispatchAll(notifications);
```

Comment:
Events must be sent outside of the transaction block. Otherwise events could be sent even when the transaction failed and was rolled back.